File: standings_game/teams.py
import itertools

def create_matchups(teams):
    return list(itertools.combinations(teams, 2))

class Team:
  def __init__(self, name, abbreviation):
    self.name = name
    self.abbreviation = abbreviation.upper()
    # stores points gained for win or draw
    self.points = 0
    # records goal difference
    self.goal_diff = 0
    # stores the results of a team's latest game
    self.single_game_points_holder = 0

mun = Team("Manchester United", "MUN", 0)
che = Team("Chelsea", "CHE", 0)
tot = Team("Tottenham Hotspurs", "TOT", 0)
liv = Team("Liverpool", "LIV", 0)
ars = Team("Arsenal", "ARS", 0)
mci = Team("Manchester City", "MCI", 0)
eve = Team("Everton", "EVE", 0)
lci = Team("Leicester", "LCI", 0)
whu = Team("West Ham United", "WHU", 0)
all_teams = [mun, che, tot, liv, ars, mci, eve, lci, whu]

File: standingsGame/teams.js
// const math = require('mathjs')
const permutations = require('./itertools');

class Team {
   constructor(name, abbreviation) {
      this.name = name;
      this.abbreviation = abbreviation.toUpperCase()
      this.points = 0;
      this.goalDiff = 0;
      this.singleGamePointsHolder = 0;
   };
};


const mun = new Team("Manchester United", "MUN")
const che = new Team("Chelsea", "CHE")
const tot = new Team("Tottenham Hotspurs", "TOT")
const liv = new Team("Liverpool", "LIV")
const ars = new Team("Arsenal", "ARS")
const mci = new Team("Manchester City", "MCI")
const eve = new Team("Everton", "EVE")
const lci = new Team("Leicester", "LCI")
const whu = new Team("West Ham United", "WHU")
const allTeams = [mun, che, tot, liv, ars, mci, eve, lci, whu]

const a = permutations.permutation(allTeams, 2)
console.log("permutations: ", Array.from(a))


// function x(args) {
//    let y = args.y ? args.y : null 
//    let z = args.z ? args.z : null

//    console.log("y: ", y)
//    console.log("z: ", z)
// }

// x({})
// x({y: 'y'})
// x({z: 'z'})
// x({y: 'y', z: 'z'})